diagnostic-settings.tf

```diff
@@ -0,0 +1,23 @@
+resource "azurerm_monitor_diagnostic_setting" "kv-ds" {
+  count = var.enable_diagnostic_setting ? 1 : 0
+
+  name                       = local.vaultName
+  target_resource_id         = azurerm_key_vault.kv.id
+  log_analytics_workspace_id = module.log_analytics_workspace.workspace_id
+
+  log {
+    category = "AuditEvent"
+
+    retention_policy {
+      enabled = true
+      days    = 14
+    }
+  }
+}
+
+module "log_analytics_workspace" {
+  count = var.enable_diagnostic_setting ? 1 : 0
```

Comment:
I would remove this we don't generally flag this sort of thing, especially for key vaults, we can add an opt out if required later

Reply:
Removed diagnostic setting toggle as suggested.